src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp

```diff
@@ -0,0 +1,275 @@
+    int64_t cameraIDkeyPoints = keyPointsInMsgBuffer.cameraID;
+
+    // stacked vector with all pixel locations of key points for 1st camera location
+    Eigen::VectorXd keyPointsStacked1(2*this->numberKeyPoints);
+    keyPointsStacked1 = cArray2EigenMatrixXd(keyPointsInMsgBuffer.keyPoints_firstImage,
+                                             2*this->numberKeyPoints,
+                                             1);
+    // convert to std vector that includes all key point pixel locations
+    this->keyPoints1.clear();
+    for (int c=0; c < this->numberKeyPoints; c++) {
+        // 2D pixel location
+        this->keyPoints1.emplace_back(keyPointsStacked1.segment(2*c, 2));
+    }
+    this->timeTag1 = keyPointsInMsgBuffer.timeTag_firstImage;
+    // dcm from inertial frame N to body frame B
+    Eigen::MRPd sigma_B1N = cArray2EigenMRPd(keyPointsInMsgBuffer.sigma_BN_firstImage);
+    Eigen::Matrix3d dcm_B1N = sigma_B1N.toRotationMatrix().transpose();
+
+    // stacked vector with all pixel locations of key points for 2nd camera location
+    Eigen::VectorXd keyPointsStacked2(2*this->numberKeyPoints);
+    keyPointsStacked2 = cArray2EigenMatrixXd(keyPointsInMsgBuffer.keyPoints_secondImage,
+                                             2*this->numberKeyPoints,
+                                             1);
+    // convert to std vector that includes all key point pixel locations
+    this->keyPoints2.clear();
+    for (int c=0; c < this->numberKeyPoints; c++) {
+        // 2D pixel location
+        this->keyPoints2.emplace_back(keyPointsStacked2.segment(2*c, 2));
+    }
+    this->timeTag2 = keyPointsInMsgBuffer.timeTag_secondImage;
+    // dcm from inertial frame N to body frame B
+    Eigen::MRPd sigma_B2N = cArray2EigenMRPd(keyPointsInMsgBuffer.sigma_BN_secondImage);
+    Eigen::Matrix3d dcm_B2N = sigma_B2N.toRotationMatrix().transpose();
+
+    /* camera config message */
+    int64_t cameraIDconfig = cameraConfigInMsgBuffer.cameraID;
+    // dcm from body frame B to camera frame C
+    Eigen::MRPd sigma_CB = cArray2EigenMRPd(cameraConfigInMsgBuffer.sigma_CB);
+    Eigen::Matrix3d dcm_CB = sigma_CB.toRotationMatrix().transpose();
+    // camera parameters
+    double alpha = 0;
+    double fieldOfView = cameraConfigInMsgBuffer.fieldOfView;
+    double resolutionX = cameraConfigInMsgBuffer.resolution[0];
+    double resolutionY = cameraConfigInMsgBuffer.resolution[1];
+    double pX = 2.*tan(fieldOfView*resolutionX/resolutionY/2.0);
+    double pY = 2.*tan(fieldOfView/2.0);
+    double dX = resolutionX/pX;
+    double dY = resolutionY/pY;
+    double up = resolutionX/2;
+    double vp = resolutionY/2;
+    // build inverse K^-1 of camera calibration matrix K
+    this->cameraCalibrationMatrixInverse << 1./dX, -alpha/(dX*dY), (alpha*vp - dY*up)/(dX*dY),
+                                            0., 1./dY, -vp/dY,
+                                            0., 0., 1.;
```

Comment:
We should consider refactoring some of these commonly used methods alongside some of the dynamics/integration refactor. For another time

Reply:
Agreed!